RedditSharp/Multi/Multi.cs

```diff
@@ -0,0 +1,362 @@
+        /// <param name="visibility">Visibility state for the Multi</param>
+        /// <param name="weightingscheme">Weighting Scheme for the Multi</param>
+        /// <param name="path">Desired URL path for the Multi</param>
+        /// <returns>A string containing the information for the newly created Multi or a status of (409) if the Multi already exists</returns>
+        public string PostMulti(string description, string displayname, string iconname, string keycolor, string[] subreddits, string visibility, string weightingscheme, string path)
+        {
+            if(Reddit.User == null)
+                throw new AuthenticationException("No user logged in");
+            var request = WebAgent.CreatePost(string.Format(GetMultiPathUrl,path));
+            var stream = request.GetRequestStream();
+            JObject modelData = new JObject();
+            modelData.Add("description_md",description);
+            modelData.Add("display_name",displayname);
+            modelData.Add("icon_name",iconname);
+            modelData.Add("key_color",keycolor);
+            JArray subData = new JArray();
+            for(int i = 0; i < subreddits.Count();i++)
+            {
+                JObject sub = new JObject();
+                sub.Add("name",subreddits[i]);
+                subData.Add(sub);
+            }
+            modelData.Add("subreddits",subData);
+            modelData.Add("visibility",visibility);
+            modelData.Add("weighting_scheme",weightingscheme);
+            WebAgent.WritePostBody(stream, new
+                {
+                    model = modelData,
+                    multipath = path,
+                    uh = Reddit.User.Modhash
+                });
+            stream.Close();
+            var response = request.GetResponse();
+            var data = WebAgent.GetResponseString(response.GetResponseStream());
+            return data;
+        }
+
+        /// <summary>
+        /// Create or update a  Multi for the authenticated user
+        /// </summary>
+        /// <param name="description">Multi Description</param>
+        /// <param name="displayname">Multi Display Name</param>
+        /// <param name="iconname">Icon Name (must be one of the default values)</param>
+        /// <param name="keycolor">Hex Code for the desired color</param>
+        /// <param name="subreddits">Array of Subreddit names to add</param>
+        /// <param name="visibility">Visibility state for the Multi</param>
+        /// <param name="weightingscheme">Weighting Scheme for the Multi</param>
+        /// <param name="path">Desired URL path for the Multi</param>
+        /// <returns>A string containing the information for the newly created or updated Multi or a status of (409) if the Multi already exists</returns>
+        public string PutMulti(string description, string displayname, string iconname, string keycolor, string[] subreddits, string visibility, string weightingscheme, string path)
```

Comment:
You may want to consider changing this to take in an object (`mData` I believe) instead of all the individual parameters for simplicity.